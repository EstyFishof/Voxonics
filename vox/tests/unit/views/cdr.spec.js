import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Filters from '@/views/Reports/CDR/components/Filters.vue';
import View from '@/views/Reports/CDR/components/View.vue';
import mock from './__mock__/cdr';
import VueTippy, { TippyComponent } from "vue-tippy";

const localVue = createLocalVue();
localVue.use(Vuex);
const router = new VueRouter();
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
// localVue.use(Vuex)

window.env = {
    BASE_URL: 'http://168.119.92.172/'
};

const paginator = {peerPage: 10, page: 1, paginator: {count: 50, recordsPerPage: 5}};

describe('Filters', () => {

    const store = new Vuex.Store({
        getters: {
            getUsersCDR: () => mock.getUsersCDR,
            getCountry: () => mock.getCountry,
            getPageCDR: () => mock.getPageCDR,
            getPeerPageCDR: () => mock.getPeerPageCDR,
            getPaginatorCDR: () => paginator
        }
    });

    const wrapper = shallowMount(Filters, { localVue, router, store });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    // it('Check snapshot', () => {
    //     expect(wrapper.element).toMatchSnapshot();
    // })
    it('clear toHaveBeenCalled', async () => {
        await wrapper.setData({paramsFilter: {user: [{name: ''}]}});
        const spy = jest.spyOn(wrapper.vm, 'clear');
        await wrapper.vm.clear();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.paramsFilter.user).toStrictEqual([]);
        spy.mockRestore();
    });
    it('setOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setOptions');
        await wrapper.vm.setOptions('endHours', '20', 'showEndHours');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.endHours).toStrictEqual('20');
        spy.mockRestore();
    });
    it('datePicker toHaveBeenCalled', async () => {
        await wrapper.setData({date: {end: '', start: ''}});
        const spy = jest.spyOn(wrapper.vm, 'datePicker');
        await wrapper.vm.datePicker();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.endHours).toStrictEqual('23');
        expect(wrapper.vm.endMin).toStrictEqual('59');
        expect(wrapper.vm.persistent).toStrictEqual(false);
        spy.mockRestore();
    });
    it('showPicker toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'showPicker');
        await wrapper.vm.showPicker();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.persistent).toStrictEqual(true);
        spy.mockRestore();
    });
    it('fetch toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'fetch');
        await wrapper.vm.fetch();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('clearDate toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'clearDate');
        await wrapper.vm.clearDate();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.startHours).toStrictEqual('23');
        expect(wrapper.vm.startMin).toStrictEqual('59');
        expect(wrapper.vm.endHours).toStrictEqual('00');
        expect(wrapper.vm.endMin).toStrictEqual('00');
        spy.mockRestore();
    });
    it('toggleVisible toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'toggleVisible');
        await wrapper.vm.toggleVisible('a');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.showList).toStrictEqual('a');
        spy.mockRestore();
    });
    it('toggleVisible toHaveBeenCalled', async () => {
        await wrapper.setData({showList: 'a'});
        const spy = jest.spyOn(wrapper.vm, 'toggleVisible');
        await wrapper.vm.toggleVisible('a');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.showList).toStrictEqual('');
        spy.mockRestore();
    });
    it('setFilterParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFilterParam');
        await wrapper.vm.setFilterParam('user', false, 'array', store.getters.getUsersCDR);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFilterParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFilterParam');
        await wrapper.vm.setFilterParam('user', store.getters.getUsersCDR[0], 'array', store.getters.getUsersCDR);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFilterParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFilterParam');
        await wrapper.vm.setFilterParam('VNumber.key', false, 'object');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFilterParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFilterParam');
        await wrapper.vm.setFilterParam('VNumber.key', 'between', 'object');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFilterParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFilterParam');
        await wrapper.vm.setFilterParam('duration.value', {from: '20'}, 'object');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFilterParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFilterParam');
        await wrapper.vm.setFilterParam('VNumber.key', 'object', 'object');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFilterParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFilterParam');
        await wrapper.vm.setFilterParam('callType', 'inbound', 'string');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('showDatePicker toHaveBeenCalled', async () => {
    //     const spy = jest.spyOn(wrapper.vm, 'showDatePicker');
    //     await wrapper.vm.showDatePicker();
    //     expect(spy).toHaveBeenCalled();
    //     expect(wrapper.vm.persistent).toStrictEqual(true);
    //     expect(wrapper.vm.isOpenDatePicker).toStrictEqual(true);
    //     spy.mockRestore();
    // })
    // it('showDatePicker toHaveBeenCalled', async () => {
    //     await wrapper.setData({isOpenDatePicker: true})
    //     const spy = jest.spyOn(wrapper.vm, 'showDatePicker');
    //     await wrapper.vm.showDatePicker();
    //     expect(spy).toHaveBeenCalled();
    //     expect(wrapper.vm.persistent).toStrictEqual(false);
    //     expect(wrapper.vm.isOpenDatePicker).toStrictEqual(false);
    //     spy.mockRestore();
    // })
    // it('computed getDate', () => {
    //     Filters.computed.getDate.get.call(wrapper.vm)
    // })
    it('computed getUserSearch', () => {
        Filters.computed.getUserSearch.get.call(wrapper.vm);
    });
    it('computed getNumbersSearch', () => {
        Filters.computed.getNumbersSearch.get.call(wrapper.vm);
    });
    it('computed getCountrySearch', () => {
        Filters.computed.getCountrySearch.get.call(wrapper.vm);
    });
    it('computed viewUsers', () => {
        Filters.computed.viewUsers.call(wrapper.vm);
    });
    it('computed viewUserNumbers', () => {
        Filters.computed.viewUserNumbers.call(wrapper.vm);
    });
    it('mounted component', async () => {
        await Filters.mounted.call(wrapper.vm);
    });
});

describe('View', () => {

    const store = new Vuex.Store({
        getters: {
            getCDR: () => mock.getCDR,
            userInfo: () => mock.userInfo,
            theme: () => mock.theme,
        }
    });

    const wrapper = shallowMount(View, { localVue, router, store });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('download toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'download');
        await wrapper.vm.download({id: '123'});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('createPlayer toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'createPlayer');
        await wrapper.vm.createPlayer(store.getters.getCDR.uniqueid);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('setTimeControl toHaveBeenCalled', async () => {
    //     const spy = jest.spyOn(wrapper.vm, 'setTimeControl');
    //     await wrapper.vm.setTimeControl(store.getters.getCDR.uniqueid);
    //     expect(spy).toHaveBeenCalled();
    //     spy.mockRestore();
    // })
    // it('setTime toHaveBeenCalled', async () => {
    //     const spy = jest.spyOn(wrapper.vm, 'setTime');
    //     await wrapper.vm.setTime(store.getters.getCDR.uniqueid);
    //     expect(spy).toHaveBeenCalled();
    //     spy.mockRestore();
    // })
    // it('jumpTime toHaveBeenCalled', async () => {
    //     const spy = jest.spyOn(wrapper.vm, 'jumpTime');
    //     await wrapper.vm.jumpTime('fw');
    //     expect(spy).toHaveBeenCalled();
    //     spy.mockRestore();
    // })
    // it('jumpTime toHaveBeenCalled', async () => {
    //     const spy = jest.spyOn(wrapper.vm, 'jumpTime');
    //     await wrapper.vm.jumpTime("back");
    //     expect(spy).toHaveBeenCalled();
    //     spy.mockRestore();
    // })
    it('setVolume toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setVolume');
        await wrapper.vm.setVolume(store.getters.getCDR.uniqueid);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('computed elapsed', () => {
        View.computed.elapsed.call(wrapper.vm);
    });
    it('computed finished', () => {
        View.computed.finished.call(wrapper.vm);
    });
    it('computed finished', () => {
        wrapper.setData({finish: 700});
        View.computed.finished.call(wrapper.vm);
    });
});
