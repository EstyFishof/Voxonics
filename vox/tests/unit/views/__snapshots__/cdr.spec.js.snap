// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Check snapshot 1`] = `
<div>
  <div
    class="wrap-table"
  >
    <table>
      <thead>
        <tr
          class="header"
        >
          <th>
            Date & Time
          </th>
           
          <th>
            Agent Name
          </th>
           
          <th>
            Agent Number
          </th>
           
          <th>
            LocalDST
          </th>
           
          <th>
            Ð¡all Status
          </th>
           
          <th>
            V-Number (DID)
          </th>
           
          <th>
            Call Type
          </th>
           
          <th>
            Phone Number
          </th>
           
          <th>
            Duration
          </th>
           
          <th>
            Bill Sec
          </th>
           
          <th>
            Cost
          </th>
           
          <th>
            Call.ID
          </th>
           
          <th>
            Listen
          </th>
           
          <th>
            Export
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr>
          <td>
            
          </td>
           
          <td>
            
          </td>
           
          <td>
            
          </td>
           
          <td>
            
          </td>
           
          <td>
            
          </td>
           
          <td>
            <img
              class="flag"
              src=""
            />
             
        
          </td>
           
          <td>
            <img
              content=""
              tabindex="0"
            />
          </td>
           
          <td>
            <span
              class="phone"
            >
              
            </span>
            <img
              class="flag"
              src=""
            />
          </td>
           
          <td>
            
          </td>
           
          <td>
            
          </td>
           
          <td>
            
          </td>
           
          <td
            content="Copy"
            tabindex="0"
          >
            <input
              class="invisible"
              type="text"
            />
             
            <img
              class="cursor"
              src="@/assets/img/icons/copy.svg"
            />
          </td>
           
          <td>
            <div>
              <audio />
               
              <img
                class="cursor disabled"
                content="Play"
                src="@/assets/img/icons/play.svg"
                tabindex="0"
              />
               
              <img
                class="cursor"
                content="Pause"
                src="@/assets/img/icons/pause.svg"
                style="display: none;"
                tabindex="0"
              />
            </div>
          </td>
           
          <td
            content="Download"
            tabindex="0"
          >
            <a>
              <img
                class="cursor disabled"
                src="@/assets/img/icons/download.svg"
              />
            </a>
          </td>
        </tr>
         
        <tr
          style="display: none;"
        >
          <td
            colspan="13"
          >
            <span
              class="player"
            >
              <span
                class="step-timer"
              >
                <img
                  class="cursor step"
                  content="-10 sec"
                  src="@/assets/img/icons/back-step.svg"
                  tabindex="0"
                  width="50"
                />
                 
                <img
                  class="cursor"
                  content="Play"
                  src="@/assets/img/icons/playBig-black.svg"
                  style="display: none;"
                  tabindex="0"
                  width="24"
                />
                 
                <!---->
                 
                <img
                  class="cursor"
                  content="Pause"
                  src="@/assets/img/icons/pauseBig.svg"
                  style="display: none;"
                  tabindex="0"
                  width="20"
                />
                 
                <img
                  class="cursor step"
                  content="+10 sec"
                  src="@/assets/img/icons/forward-step.svg"
                  tabindex="0"
                  width="50"
                />
              </span>
               
              <span
                class="timer"
              >
                <span>
                  0:00
                </span>
                 
                <span
                  class="range-wrapper timer-width"
                >
                  <input
                    class="time"
                    max="100"
                    step="1"
                    type="range"
                  />
                </span>
                 
                <span>
                  0:00
                </span>
              </span>
               
              <span
                class="volume-wrapper"
              >
                <img
                  alt=""
                  class="hover"
                  src="@/assets/img/icons/quiet.svg"
                />
                 
                <img
                  alt=""
                  class="mute hover"
                  src="@/assets/img/icons/mute.svg"
                  style="display: none;"
                />
                 
                <span
                  class="range-wrapper"
                >
                  <input
                    class="volume"
                    max="1"
                    step="0.01"
                    type="range"
                  />
                </span>
                 
                <img
                  alt=""
                  class="hover"
                  height="24"
                  src="@/assets/img/icons/volumeUp.svg"
                />
              </span>
            </span>
          </td>
           
          <td>
            <span
              class="up"
              content="Close player"
              tabindex="0"
            >
              <svg
                fill="none"
                height="15"
                viewBox="0 0 13 13"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M8.03177 6.50013L12.6823 1.84948C13.1059 1.42591 13.1059 0.741255 12.6823 0.317681C12.2588 -0.105894 11.5741 -0.105894 11.1506 0.317681L6.5 4.96833L1.84944 0.317681C1.42588 -0.105894 0.74124 -0.105894 0.317674 0.317681C-0.105891 0.741255 -0.105891 1.42591 0.317674 1.84948L4.96823 6.50013L0.317674 11.1508C-0.105891 11.5744 -0.105891 12.259 0.317674 12.6826C0.528915 12.8938 0.806237 13 1.08356 13C1.36088 13 1.6382 12.8938 1.84944 12.6826L6.5 8.03194L11.1506 12.6826C11.3618 12.8938 11.6391 13 11.9164 13C12.1938 13 12.4711 12.8938 12.6823 12.6826C13.1059 12.259 13.1059 11.5744 12.6823 11.1508L8.03177 6.50013Z"
                  fill="#515960"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
